### PR TITLE
Creation of intersection utilities (together with tests)

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR adds to the core a new utility for computing intersections (so far line-line in 2D and triangle-line in 3D). It is aimed to be used in the discontinuous distance computation to count how many edges are intersected and retrieve the intersection points to compute the least squares fitting plane.

### CODE REVIEW

File: kratos/tests/utilities/test_intersection_utilities.cpp [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+            line_pt_1.Coordinates(), 
+            line_pt_2.Coordinates(), 
+            int_pt.Coordinates());
+
+        // Compute and check the obtained intersection point coordinates
+        const array_1d<double,3> int_pt_coords = int_pt.Coordinates();
+        KRATOS_CHECK_EQUAL(int_id, 1);
+        KRATOS_CHECK_NEAR(int_pt_coords(0), 0.4, 1e-6);
+        KRATOS_CHECK_NEAR(int_pt_coords(1), 0.6, 1e-6);
+        KRATOS_CHECK_NEAR(int_pt_coords(2), 0.0, 1e-6);
```
Comment (RiccardoRossi) [inline kratos/tests/utilities/test_intersection_utilities.cpp]: i would use a stricterbprecision here

**Review by RiccardoRossi — DISMISSED** [review]

**Comment by rubenzorrilla** [comment]
@RiccardoRossi done! (1e-10 should be precise enough).

**Review by RiccardoRossi — DISMISSED** [review]

File: kratos/utilities/intersection_utilities.h [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_INTERSECTION_TRIANGLE_LINE_UTILITY )
```
Comment (pooyan-dadvand) [inline kratos/utilities/intersection_utilities.h]: Is not the same of the file name

**Review by pooyan-dadvand — COMMENTED** [review]
I have not reviewed the algorithm yet but epsilon should be configurable

**Review by pooyan-dadvand — COMMENTED** [review]
I have not reviewed the algorithm yet but epsilon should be configurable

File: kratos/utilities/intersection_utilities.h (hunk 2 above)
Comment (rubenzorrilla) [reply kratos/utilities/intersection_utilities.h]: My bad. I changed the file name during the implementation.

**Comment by rubenzorrilla** [comment]
@pooyan-dadvand . Do we want a user-defined or a hardcoded value for epsilon?

**Comment by pooyan-dadvand** [comment]
I would go for the epsilon as a member of class and configurable  but not as a new argument to the functions

**Comment by RiccardoRossi** [comment]
@pooyan-dadvand these are static functions... we cannot set it in the class.

I would use a epsilon parameter and assign a default value to it

**Comment by rubenzorrilla** [comment]
I've set std::numeric_limits<double>::epsilon() as default value for epsilon parameter. Is it ok?

**Comment by pooyan-dadvand** [comment]
Why you want them to be static? 
These methods already had 4 arguments ....

**Comment by rubenzorrilla** [comment]
Because they are intended to be called in other functionalities (e.g. the distance computation). I thought it was a good idea to implement them as static.

**Review by RiccardoRossi — DISMISSED** [review]

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**